TSOClient/tso.client/UI/Panels/UIChatDialog.cs

```diff
@@ -114,6 +122,39 @@ public override void Draw(UISpriteBatch batch)
             base.Draw(batch);
         }
 
+        public void ResizeChatDialogByFactor(int factor)
+        {
+            if (chatSize + factor < 7 || chatSize + factor > 11)
+            {
+                return;
+            }
+            chatSize += factor;
+
+            var updatedSize = factor * 30; //Incease or decrease by 30px
+
+            var histStyle = ChatHistoryText.TextStyle.Clone();
+            histStyle.setIdealFontSize(chatSize);
+            ChatHistoryText.TextStyle = histStyle;
+
+            this.SetSize((int)Math.Round(this.Size.X) + updatedSize, (int)Math.Round(this.Size.Y) + updatedSize);
+
+            ChatHistoryBackground.SetSize(ChatHistoryBackground.Size.X + updatedSize, ChatHistoryBackground.Size.Y + updatedSize);
+            ChatHistoryText.SetSize(ChatHistoryBackground.Size.X - 19, ChatHistoryBackground.Size.Y - 16);
+
+            ChatHistorySlider.Position = new Vector2(ChatHistorySlider.Position.X + updatedSize, ChatHistorySlider.Position.Y);
+            ChatHistorySlider.SetSize(ChatHistorySlider.Size.X, ChatHistoryBackground.Size.Y - 26);
+            ChatHistoryScrollUpButton.Position = new Vector2(ChatHistoryScrollUpButton.Position.X + updatedSize, ChatHistoryScrollUpButton.Position.Y);
+            ChatHistoryScrollDownButton.Position = new Vector2(ChatHistoryScrollDownButton.Position.X + updatedSize, ChatHistoryScrollDownButton.Position.Y + updatedSize);
+
+            ChatEntryTextEdit.Position = new Vector2(ChatEntryTextEdit.Position.X, ChatEntryTextEdit.Position.Y + updatedSize);
+            ChatEntryTextEdit.SetSize(ChatEntryTextEdit.Size.X, ChatEntryTextEdit.Size.Y);
+            ChatEntryBackground.Position = new Vector2(ChatEntryBackground.Position.X, ChatEntryBackground.Position.Y + updatedSize);
+            ChatEntryBackground.SetSize(ChatEntryBackground.Size.X + updatedSize, ChatEntryBackground.Size.Y);
```

Comment:
Resizing the window with the text size is a good idea. In future, it might be useful to be able to resize this window by dragging on the edges. I believe it was the only window you could do that on in tso.

Reply:
Yeah in the future I'd like to make this even smarter, I was unsure if this project would even accept pull requests, so I put in a version that is kinda the best of both worlds for the initial version, to see what the reception from developers to users would be.

Reply:
I would love to implement the re-size window feature in the near future.